### ISSUE TITLE
curl - add more protocols and features

### ISSUE THREAD

**Issue by takov** [clearlinux/distribution#1969]
I am with ClearLinux 33090 and curl version: 7.70.0. Currently curl supports the following protocols and features:
**Protocols: dict file ftp ftps http https imap imaps rtsp smtp smtps 
Features: AsynchDNS HTTP2 HTTPS-proxy IPv6 Largefile libz SSL TLS-SRP UnixSockets**

curl from other distros is enabled to supports more protocols:
**Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp 
Features: AsynchDNS IDN IPv6 Largefile GSS-API Kerberos SPNEGO NTLM NTLM_WB SSL libz TLS-SRP HTTP2 UnixSockets HTTPS-proxy PSL** 

It would be really grate if not all but at least:
- sftp
- scp 
- ldap
-GSS-API
- Kerberos
-  HTTPS-proxy
- NTLM
are added to curl.

**Comment by ahkok** [comment]
Yes and ... maybe no. `curl` is part of our core bundle and adding some of these protocols would mean a sizeable increase of our base OS. I don't think we can make this pluggable, but, that is the main reason why we haven't enabled some of these yet.

**Comment by takov** [comment]
Thank you for the quick response. 
It make sense.

**Comment by thiagomacieira** [comment]
@ahkok do we need curl or libcurl?

If it's libcurl, we can have two libraries, one in a fallback search path in the core OS, the second with more features in case the user wants to.

If it's the binary, we'd have to make one exec() the other if it's present, like we've done for emacs/emacs-x11, usrbinpython, etc.

**Comment by ahkok** [comment]
we need `libcurl` in `os-core`. We can absolutely make a second `curl` that has all the fruits and vegetables.

**Comment by thiagomacieira** [comment]
It's likely that curl itself doesn't have features, that they come from libcurl. So we'd need two libs like I outlined. That probably means modifying the dynamic linker to have an extra fallback path we can install minimal libs to.

**Comment by ahkok** [comment]
Note: `HTTPS-proxy` is supported, it's in the list of features that curl outputs.

**Comment by ahkok** [comment]
The other items you're requesting are indeed something we want to keep out of `os-core` and thus this will be not trivial.
